proposals/reward-proposal-v2-reward-calculation.md

```diff
@@ -0,0 +1,109 @@
+---
+title: Epoch Reward Calculation
+---
+
+## Problem
+
+Calculating and distributing rewards at the first slot on the epoch boundary is
+not scalable. With increasing number of stake accounts and vote accounts on
+validator nodes, the block time at epoch boundary can last as long as 20 seconds
+because of paying the epoch rewards. Such a long block time is undesirable and
+should be avoided.
+
+To solve this problem and speed up the reward time at epoch boundary, an earlier
+approach [Partitioned Inflationary Rewards
+Distribution](https://github.com/solana-labs/solana/blob/master/docs/src/proposals/partitioned-inflationary-rewards-distribution.md)
+is proposed. Let's call this approach `V1`. A prototype of the `V1` proposal is
+implemented. While the `V1` approach works functionally, it, however, has two
+main drawbacks.
+
+First, a restriction is enforced to prevent updates to all stake accounts and
+vote accounts during reward interval. In other words, the stake accounts and
+vote accounts are **locked** during reward interval. All stake account's actions
+such as delegate, redelegate, withdraw etc. are disallowed during reward
+interval and have to wait after reward interval.
+
+Second, there is no double-entry booking available for epoch rewards. Currently,
+epoch rewards are rewarded and recorded inside the first block of the epoch.
+Lack of double-entry booking is less a problem for the current approach, because
+all rewards are contained one block - the first block of the epoch, which makes
+block reward verification somewhat easier and trackable. However, with
+distributed rewards (such as the `V1` approach ), lack of double-entry booking
+becomes a more serious limitation. Since the rewards distribution can span
+multiple blocks, it becomes more difficult to verify the correctness of the
+reward payment without the ability for double-entry booking.
+
+Given these two drawbacks above, we are proposing an improved way to handle
+epoch reward - `V2`. The `V2` approach is based on `V1` but with the
+improvements to address these two drawbacks.
+
+To make the discussion clearer, the `V2` approach is split into two proposals:
+`rewards calculation` and `rewards distribution`. In this proposal, we will
+focus on "reward calculation". A follow-up proposal will talk about "reward
+distribution".
+
+## Proposed Solutions
+
+Unlike the current epoch rewards computation approach, in which computing the
+rewards happens at the first slot of epoch boundary in the bank runtime
+directly, the new reward computation is moved to a background service in the
+proposed approach.
+
+This is very similar to the `V1` proposal, a separate service,
+"EpochRewardCalculationService" is created at the start time of the validator.
+This service is listening on a channel for incoming rewards calculation
+requests, and handle the requests to compute the rewards.
```

Comment:
Not a super big deal but in general SIMDs don't need to detail how a proposal would be implemented in a particular validator code base. Firedancer would likely be implemented differently. 

This paragraph could be made more clear if it started with "As an example, in the Solana Labs validator client a separate service "EpochRewardCalculationService" is created ..."